cloudinit/sources/helpers/azure.py

```diff
@@ -177,6 +185,46 @@ def report_diagnostic_event(str):
     return evt
 
 
+def report_compressed_event(event_name, event_content):
+    """Report a compressed event"""
+    compressed_data = base64.encodebytes(zlib.compress(event_content))
+    event_data = {"encoding": "gz+b64",
+                  "data": compressed_data.decode('ascii')}
+    evt = events.ReportingEvent(
+        COMPRESSED_EVENT_TYPE, event_name,
+        json.dumps(event_data),
+        events.DEFAULT_EVENT_ORIGIN)
+    events.report_event(evt, excluded_handlers={"logging"})
```

Comment:
I think this exclude_handlers might need to change. The `reportining:` keys are just opaque strings that have no real meaning, on certain images the top-level key in that reporting dict could be called "logging" and others the top-level key could be "mylogs".
our default is "logging" as you saw, but what matters  to your reporting branch is that "logging" happens to be of "type: log" versus "type: hyperv". 

```
reporting:
   logging:
     type: log
```
So, I think that we would want to make this param "excluded_handler_types=["log"]) " and change the logic on excluded_handlers a bit to check each registered handler to see if it is an instance of the excluded handler type.

Reply:
one way we could do that is something remote like this (untested):

```
diff --git a/cloudinit/reporting/events.py b/cloudinit/reporting/events.py
index 55313769f..face568f6 100644
--- a/cloudinit/reporting/events.py
+++ b/cloudinit/reporting/events.py
@@ -12,7 +12,7 @@ import base64
 import os.path
 import time
 
-from . import instantiated_handler_registry
+from . import instantiated_handler_registry, available_handlers
 
 FINISH_EVENT_TYPE = 'finish'
 START_EVENT_TYPE = 'start'
@@ -94,10 +94,18 @@ def report_event(event, excluded_handlers=None):
         The type of the event; this should be a constant from the
         reporting module.
     """
+    if not excluded_handler_types:
+        excluded_handler_types = []
+    excluded_handler_classes = [
+        hndl_cls
+        for hndl_type, hndl_cls in available_handlers.registered_items.items()
+        if hndl_type in excluded_handler_types
+    ]
     handlers = instantiated_handler_registry.registered_items.items()
-    for handler_key, handler in handlers:
-        if excluded_handlers and handler_key in excluded_handlers:
-            continue    # skip this excluded handler
+    for handler_name, handler in handlers:
+        for excluded_cls in excluded_handler_classes:
+            if isinstance(handler, excluded_cls):
+                continue    # skip this excluded handler
         handler.publish_event(event)
```